ocpmodels/preprocessing/atoms_to_graphs.py

```diff
@@ -0,0 +1,186 @@
+import ase.db
+import numpy as np
+from pymatgen.io.ase import AseAtomsAdaptor
+
+import torch
+from ocpmodels.common.utils import collate
+from ocpmodels.datasets.elemental_embeddings import EMBEDDINGS
+from torch_geometric.data import Data
+
+try:
+    shell = get_ipython().__class__.__name__
+    if shell == "ZMQInteractiveShell":
+        from tqdm.notebook import tqdm
+    else:
+        from tqdm import tqdm
+except NameError:
+    from tqdm import tqdm
+
+
+class AtomsToGraphs:
+    def __init__(
+        self, max_neigh=12, radius=6, dummy_distance=7, dummy_index=-1
+    ):
+        self.max_neigh = max_neigh
+        self.radius = radius
+        self.dummy_distance = dummy_distance
+        self.dummy_index = dummy_index
+        self.data_list = []
+
+    def _get_neighbors_pymatgen(self, atoms):
+        self.atoms = atoms
+        self.struct = AseAtomsAdaptor.get_structure(self.atoms)
+        # these return jagged arrays meaning certain atoms have more neighbors than others
+        _c_index, n_index, _images, n_distance = self.struct.get_neighbor_list(
+            r=self.radius
+        )
```

Comment:
I see; so this is the primary diff that makes it much faster!

Reply:
Yeah, I vectorized a few things and removed one unnecessary loop over the data, but this was largely responsible for the speed up